toolkit/tools/pkg/imagecustomizerlib/configvalidation.go

```diff
@@ -625,3 +669,56 @@ func resolveKernelCommandLine(configChain []*ConfigWithBasePath) imagecustomizer
 		ExtraCommandLine: mergedArgs,
 	}
 }
+
+func resolveStorage(configChain []*ConfigWithBasePath) imagecustomizerapi.Storage {
+	var resolvedStorage imagecustomizerapi.Storage
+
+	for _, configWithBase := range slices.Backward(configChain) {
+		storage := configWithBase.Config.Storage
+
+		if len(storage.Disks) > 0 {
+			// Override
+			resolvedStorage.Disks = storage.Disks
+			resolvedStorage.BootType = storage.BootType
+			resolvedStorage.FileSystems = storage.FileSystems
+			resolvedStorage.Verity = storage.Verity
+			return resolvedStorage
+		}
+
+		// .storage.resetPartitionsUuidsType - override
+		if storage.ResetPartitionsUuidsType != imagecustomizerapi.ResetPartitionsUuidsTypeDefault {
+			resolvedStorage.ResetPartitionsUuidsType = storage.ResetPartitionsUuidsType
```

Comment:
If you are iterating backwards through the chain, you need to use early return. Otherwise, you will pick the first value in the chain instead of the last.

That being said, I think the storage merging logic is complex enough that it would probably be better to merge storage using recursion on the config tree. (The other fields have simple merging logic. So, going backwards through the linear chain works fine.)